src/main/java/com/kapeta/spring/rabbitmq/KapetaRabbitConnectionFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2023 Kapeta Inc.
+ * SPDX-License-Identifier: MIT
+ */
+package com.kapeta.spring.rabbitmq;
+
+import com.rabbitmq.client.ConnectionFactory;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.amqp.AmqpIOException;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.support.ListenerExecutionFailedException;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.retry.support.RetryTemplateBuilder;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.kapeta.spring.rabbitmq.RabbitHelper.defaultRetryTemplate;
+
+@Slf4j
+public class KapetaRabbitConnectionFactory {
+
+    public static final String PORT_AMQP = "amqp";
+
+    private static final int MAX_ATTEMPTS = 100;
```

Comment:
Should this be configurable?

Reply:
It's only used for an internal ensure vhost thing so figured it'd be something people wont change